Lesson04

kiseleva1506 opened the pull request.

kiseleva1506 committed:
task 1

kiseleva1506 committed:
task 2

kiseleva1506 committed:
task 3

kiseleva1506 committed:
task 4

kiseleva1506 committed:
task 5

kiseleva1506 committed:
task 6

kiseleva1506 committed:
task 7

GefMar commented on homeworks/lesson04/task1.py:
```diff
@@ -0,0 +1,35 @@
+    i = 0
+    while i < len(func_args):
+        try:
+            func_args[i] = float(func_args[i])
```
так как нам надо все параметры преобразовать можно обойтись подобной конструкцией: 

```
try:
    func_args = [float(itm) for itm in func_args]
except ValueError:
    print('ошибка ввода')
    exit()
```

и обойтись без while и этой громоздкой конструкции. 

Нам не важно когда возникнет ошибка, важ ен факт того что все данные без ошибки

GefMar commented on homeworks/lesson04/task2.py:
```diff
@@ -0,0 +1,12 @@
+Результат: [12, 44, 4, 10, 78, 123]. """
+
+my_list1 = [300, 2, 12, 44, 1, 1, 4, 10, 7, 1, 78, 123, 55]
+my_list2 = [my_list1[i] for i in range(1, len(my_list1)) if my_list1[i] > my_list1[i - 1]]
```
а можно было сделать не через range, а enumerate()

GefMar commented on homeworks/lesson04/task5.py:
```diff
@@ -0,0 +1,10 @@
+всех элементов списка.
+Подсказка: использовать функцию reduce(). """
+
+from functools import reduce
```
в 3йм уроке было пожелание реализовать свой reduce - попробуйте он не сложный

GefMar commented on homeworks/lesson04/task7.py:
```diff
@@ -0,0 +1,18 @@
+
+def fact(n):
+    tmp = 1
+    for i in range(1, n + 1):
```
попробуйте решить без range

kiseleva1506 committed:
Исправления по результатам проверки
task1 проверка типа параметров в генераторе
task2 добавила решение через enumerate
task5 добавила my_reduce
task7 факториал без использования range

kiseleva1506 committed:
Убрано сравнение счетчика с нулем, т.е. любое значение, отличное от нуля, считается True
